src/spec/models/function_spec.rb

```diff
@@ -1,16 +1,59 @@
 require 'rails_helper'
 
+
 RSpec.describe Function, type: :model do
   subject { create :function }
 
-  it { is_expected.to validate_presence_of :name }
-  it { is_expected.to validate_uniqueness_of(:name).scoped_to(:user_id) }
-  it { is_expected.to allow_value('function_name').for :name }
-  it { is_expected.to allow_value('functionName').for :name }
-  it { is_expected.to allow_value('function1').for :name }
-  it { is_expected.to_not allow_value('name with space').for :name }
-  it { is_expected.to_not allow_value('name_with\backslash').for :name }
-  it { is_expected.to_not allow_value('اسم_عربي').for :name }
+  def generate_name (code)
+    subject.code = code
+    subject.validate
+  end
+
+  # it { is_expected.to allow_value('function_name').for :name }
+  describe '#name' do
+    it 'must exist' do
+      generate_name 'def;end'
+      expect(subject.errors[:name]).to be_present
+    end
+
+
+    # it { is_expected.to validate_uniqueness_of(:name).scoped_to(:user_id) }
+    it 'must be unique' do
```

Comment:
I think using shoulda-matchers will be better

Reply:
Doesn't work here because 'name' is generated from 'code' so it will always fail.